stockflux-securities-master/src/components/securities-table/SecuritiesTable.js

```diff
@@ -0,0 +1,133 @@
+import React, { useEffect, useCallback, useReducer } from "react";
+import Components from "stockflux-components";
+import { Link } from "react-router-dom";
+import "./SecuritiesTable.css";
+import * as service from "../../services/SecuritiesService";
+import ValidationError from "../../services/ValidationError";
+import Alert, { AlertType } from "../alert/Alert";
+import { TableState } from "../../enums";
+import Button, { ButtonSize } from "../button/Button";
+import {
+  securitiesTableReducer,
+  initialTableState
+} from "../../reducers/securitiesTableReducer";
+import {
+  tableLoading,
+  tableUpdating,
+  tableError,
+  tableSuccess,
+  setSecuritiesData
+} from "../../actions/securitiesTableActions";
+import PropTypes from "prop-types";
+import classNames from "classnames";
+import TableBody from "./TableBody";
+
+const SecuritiesTable = ({ location }) => {
+  const [state, dispatch] = useReducer(
+    securitiesTableReducer,
+    initialTableState
+  );
+
+  const handleError = err => {
+    const message =
+      err instanceof ValidationError ? err.messages : [err.message];
+    dispatch(tableError(message));
+  };
+
+  const loadSecurities = useCallback(messages => {
+    service
+      .getSecuritiesData()
+      .then(securities => {
+        dispatch(setSecuritiesData(securities));
+        dispatch(tableSuccess(messages));
+      })
+      .catch(err => {
+        dispatch(setSecuritiesData([]));
+        handleError(err);
+      });
+  }, []);
+
+  useEffect(() => {
+    dispatch(tableLoading());
+    loadSecurities(!!location.state ? location.state.messages : []);
+  }, [loadSecurities, location.state]);
+
+  const deleteSecurity = securityId => {
+    dispatch(tableUpdating());
+    service
+      .deleteSecurity(securityId)
+      .then(() => {
+        loadSecurities(["Security Successfully Deleted"]);
+      })
+      .catch(handleError);
+  };
+
+  const patchSecurity = (securityId, updates) => {
+    dispatch(tableUpdating());
+    service
+      .patchSecurity(securityId, updates)
+      .then(() => {
+        loadSecurities(["Security Updated"]);
+      })
+      .catch(handleError);
+  };
+
+  return (
+    <div className="securities-container">
+      <div className="securities-header-container">
+        <h1 className="securities-title">Securities</h1>
+        <div className="add-securities-button-above-table">
+          {state.securitiesData.length > 0 && (
+            <Link to="/inputform">
+              <Button size={ButtonSize.SMALL} className="add-securities-button">
+                Add Security
+              </Button>
+            </Link>
+          )}
+        </div>
+      </div>
+      <div className="heading-container">
+        <h2 className="securities-table-heading">Exchange</h2>
+        <h2 className="securities-table-heading">Symbol</h2>
+        <h2 className="securities-table-heading">Name</h2>
+        <h2 className="securities-table-heading">Edit / Delete</h2>
```

Comment:
You could also use map here, e.g. something similar to this:
```
['Exchange', 'Symbol', 'Name', 'Edit / Delete'].map(heading => (<h2 className="securities-table-heading">{heading}</h2>)
```

It might save a couple of lines and reduce repetition. It's up to you though :)

Reply:
Good idea, I've done this now and changed `Edit / Delete` to `Options` as there are now more than just the two options.